test/blackbox-tests/test-cases/pkg/ocamlformat/binary-avoid-from-project-deps.t

```diff
@@ -0,0 +1,164 @@
+Make sure that $ dune fmt does not take the OCamlFormat binary inside the "dune-project"
+dependencies.
+
+  $ . ../helpers.sh
+  $ mkrepo
+
+Make a fake ocamlformat.0.26.2:
+  $ mkdir ocamlformat
+  $ cd ocamlformat
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package (name ocamlformat))
+  > EOF
+  $ cat > ocamlformat.ml <<EOF
+  > let version = "0.26.2"
+  > ;;
+  > let () = print_endline ("formatted with version "^version)
+  > EOF
+  $ cat > dune <<EOF
+  > (executable
+  >  (public_name ocamlformat))
+  > EOF
+  $ cd ..
+  $ tar -czf ocamlformat-0.26.2.tar.gz ocamlformat
+
+Make a fake ocamlformat.0.26.3:
+  $ cd ocamlformat
+  $ cat > ocamlformat.ml <<EOF
+  > let version = "0.26.3"
+  > let () = print_endline ("formatted with version "^version)
+  > EOF
+  $ cd ..
+  $ tar -czf ocamlformat-0.26.3.tar.gz ocamlformat
+  $ rm -rf ocamlformat
+
+Make a package for the fake OCamlformat library:
+  $ mkpkg ocamlformat 0.26.2 <<EOF
+  > build: [
+  >   [
+  >     "dune"
+  >     "build"
+  >     "-p"
+  >     name
+  >     "@install"
+  >   ]
+  > ]
+  > url {
+  >  src: "file://$PWD/ocamlformat-0.26.2.tar.gz"
+  >  checksum: [
+  >   "md5=$(md5sum ocamlformat-0.26.2.tar.gz | cut -f1 -d' ')"
+  >  ]
+  > }
+  > EOF
+
+Make a package for the fake OCamlformat library:
+  $ mkpkg ocamlformat 0.26.3 <<EOF
+  > build: [
+  >   [
+  >     "dune"
+  >     "build"
+  >     "-p"
+  >     name
+  >     "@install"
+  >   ]
+  > ]
+  > url {
+  >  src: "file://$PWD/ocamlformat-0.26.3.tar.gz"
+  >  checksum: [
+  >   "md5=$(md5sum ocamlformat-0.26.3.tar.gz | cut -f1 -d' ')"
+  >  ]
+  > }
+  > EOF
+
+Make a project that uses the fake ocamlformat:
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package
+  >  (name foo)
+  >  (depends (ocamlformat (= 0.26.2))))
+  > EOF
+  $ cat > foo.ml <<EOF
+  > let () = print_endline "Hello, world"
+  > EOF
+  $ cat > dune <<EOF
+  > (executable
+  >  (public_name foo))
+  > EOF
+  $ cat > dune-workspace <<EOF
+  > (lang dune 3.13)
+  > (lock_dir
+  >  (path "dev-tools.locks/ocamlformat")
+  >  (repositories mock))
+  > (lock_dir
+  >  (repositories mock))
+  > (repository
+  >  (name mock)
+  >  (source "file://$(pwd)/mock-opam-repository"))
+  > EOF
+
+Add a fake binary in the PATH
```

Comment:
Was this left in by mistake? This test just makes sure that `dune fmt` doesn't invoke the `ocamlformat` executable from the project's package dependencies, not from PATH.

Reply:
Sure, but at no point in the current file is the ocamlformat executable in PATH actually run. What's the point of adding it to PATH?

Reply:
Yeah I will remove it.